test/e2e/virtual/initializingworkspaces/virtualworkspace_test.go

```diff
@@ -414,25 +411,49 @@ func TestInitializingWorkspacesVirtualWorkspaceAccess(t *testing.T) {
 		require.Empty(t, cmp.Diff(configMaps.Items, []corev1.ConfigMap{}))
 	}
 
+	patchBytesFor := func(ws *tenancyv1alpha1.ClusterWorkspace, mutator func(*tenancyv1alpha1.ClusterWorkspace)) []byte {
```

Comment:
generic helper. Put it at the bottom or even into the framework. Can work on every runtime.Object with `ws.DeepCopyObject()` method.

Reply:
This will be a TODO